src/python/WMCore/MicroService/Unified/MSOutput.py

```diff
@@ -593,11 +612,56 @@ def docKeyUpdate(self, msOutDoc, **kwargs):
                 self.logger.warning(msg)
         return msOutDoc
 
-    def docInfoUpdate(self, msOutDoc):
+    def docInfoUpdate(self, msOutDoc, pipeLine=None):
         """
         A function intended to fetch and fill into the document all the needed
         additional information like campaignOutputMap etc.
         """
+
+        # Fill the destinationOutputMap first
+        if msOutDoc['isRelVal']:
+            destinationOutputMap = []
+            wflowDstSet = set()
+            updateDict = {}
+            for dataset in msOutDoc['OutputDatasets']:
+                _, dsn, procString, dataTier = dataset.split('/')
+                destination = set()
+                if dataTier != "RECO" and dataTier != "ALCARECO":
+                    destination.add('T2_CH_CERN')
+                if dataTier == "GEN-SIM":
+                    destination.add('T1_US_FNAL_Disk')
+                if dataTier == "GEN-SIM-DIGI-RAW":
+                    destination.add('T1_US_FNAL_Disk')
+                if dataTier == "GEN-SIM-RECO":
+                    destination.add('T1_US_FNAL_Disk')
+                if "RelValTTBar" in dsn and "TkAlMinBias" in procString and dataTier != "ALCARECO":
+                    destination.add('T2_CH_CERN')
+                if "MinimumBias" in dsn and "SiStripCalMinBias" in procString and dataTier != "ALCARECO":
+                    destination.add('T2_CH_CERN')
+                dMap = {'datasets': [dataset],
+                        'destination': list(destination)}
+                destinationOutputMap.append(dMap)
```

Comment:
As we discussed over slack, this will add one dict per dataset, while we should aggregate datasets going to the same destination. Change should be fairly simple, so I'd rather have it implemented in this PR.

Reply:
Done